elasticsearch/src/main/java/org/hibernate/search/elasticsearch/schema/impl/model/IndexMetadata.java

```diff
@@ -20,17 +21,17 @@
  */
 public class IndexMetadata {
 
-	private String name;
+	private URLEncodedString name;
```

Comment:
yes I agree with that and I actually was tempted to do it.. but for sake of getting this done I stopped polishing :)

+1 to do such a thing at another round